Phalanx/Assets/Scripts/Entity/Unit.cs

```diff
@@ -7,43 +7,86 @@ namespace Olympus.Phalanx.Entity
 {
     public class Unit : MonoBehaviour, IOccupant
     {
+        //need time since last moved.
+        //Need destination Tile .
 
+        private float timeStartedMovement;
+        private float distanceToNextTile;
+        private bool canMove;
         // Use this for initialization
         void Start()
         {
-
         }
 
         // Update is called once per frame
         void Update()
         {
+            if (canMove)
+                move();
+        }
 
+        public void move()
+        {
+            //Why does is it giving me null reference? I break pointed to it and clearly is there and correct val
+            if (transform.position != destination.transform.position)
```

Comment:
You probably need to make a check for if destination is a valid gameObject
Possibly just make destination a Vector3 in it of itself.